### DISCUSSION TITLE
Javascript access to DOM

### DISCUSSION THREAD

**Discussion by nunopoco** [AutomaApp/automa#1067] [Q&A] [answered]
Hi,
I need help with running JS on Automa.

`const xpath = "(//*[contains(text(), '{1}')])[1]";
var elem = document.evaluate(xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
elem.textContent = "test";`

This code runs in dev tools but not using Automa.
The website allows code injections so it's not that.
TIA

**Comment by Kholid060** [comment, marked as answer]
Have you checked if there's an error on the log of the workflow or the console tab of the Chrome Dev tools after running the workflow?

**Reply by nunopoco** [reply]
No error, I get a success message and a "..." output on the console from the `https://static.canva.com/static/lib/sentry/6.7.1.min.js` file.
Also, there's a warning on page load but I don't know if it has anything to do with this extension, the website or another extension:
`invoking a transformer from outside a reactive context won't memorized and is cleaned up immediately, unless keepAlive is set`

**Reply by nunopoco** [reply]
Any tips on how I should do this @Kholid060? 🙏

**Reply by Kholid060** [reply]
Sorry for the late reply, can you send me the workflow? So I can test it.

**Reply by nunopoco** [reply]
Had to put it in a zip folder as github doens't allow a json file as attachment:
[automa.zip](https://github.com/AutomaApp/automa/files/10424946/automa.zip)

It's very simple.
I'm using Canva and trying to detect if there's a text with the string "{1}" and if there is, it gets changed to "test".

Thanks @Kholid060

**Reply by Kholid060** [reply]
I think it's working because the XPath matches the script element that Automa injected
![image](https://user-images.githubusercontent.com/22908993/212798993-016cfe89-4aaf-4c42-91ff-ef9f2fa57f1a.png)

You can fix it by using this selector `(//*[contains(text(), '{1}')][not(self::script)])[1]`

**Reply by nunopoco** [reply]
It works, thanks @Kholid060.
Didn't know about this Xpath instruction to exclude searching in the script tags. 
I guess I never used it that way and expected it to only search inside HTML elements.
